cidr-filtering-cni

```diff
@@ -0,0 +1,252 @@
+}
+
+exitWithSuccess() {
+    prev_result=$(get_object "${stdin}" ".prevResult")
+    if [[ "$prev_result" == "null" ]]; then
+        echo "{"cniVersion": "$(jq -r ".cniVersion" <<< "$stdin")"}" >&3
+    else
+        echo "$prev_result" >&3
+    fi
+    exit 0
+}
+
+create_chains_for_filtering() {
+    local policy_id="$1"
+    local table_type="$2"
+    local direction="$3"
+
+    local match_iface="iifname"
+    if [[ "$direction" == "egress" ]]; then
+        local match_iface="oifname"
+    fi
+
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}") || exitWithError "Failed to add chain m-${policy_id}-${direction}"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${direction}" "${match_iface}" "$CNI_IFNAME" counter jump m-"${policy_id}"-"${direction}") || exitWithError "Failed to add rule: jump m-${policy_id}-${direction}"
+
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}"-subnets) || exitWithError "Failed to add chain m-"${policy_id}"-"${direction}"-subnets"
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}"-ports) || exitWithError "Failed to add chain m-"${policy_id}"-"${direction}"-ports"
+    
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}" counter jump m-"${policy_id}"-"${direction}"-subnets) || exitWithError "Failed to add rule: jump m-"${policy_id}"-"${direction}"-subnets"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}" counter jump m-"${policy_id}"-"${direction}"-ports) || exitWithError "Failed to add rule: jump m-"${policy_id}"-"${direction}"-ports"
+
+    create_rules_for_filtering "${policy_id}" "${table_type}" "${direction}" "${match_iface}"
+}
+
+case $CNI_COMMAND in
+ADD)
+    echo "CNI_NETNS: $CNI_NETNS" >> $logFile
+    echo "CNI_CONTAINERID: $CNI_CONTAINERID" >> $logFile
+    echo "STDIN: $stdin" >> $logFile
+
+    policy_name=$(echo "$stdin" | jq -r ".name")
+    policy_id="$(echo "$policy_name" | sha1sum )"
+    policy_id="${policy_id:0:5}" # use first 5 characters to identify a policy
+
+    mkdir -p /var/run/netns/
+    ln -sfT "$CNI_NETNS" /var/run/netns/"${CNI_CONTAINERID}"
+
+    ip netns exec "$CNI_CONTAINERID" $(create_table ${NFT_TABLE_BRIDGE} ${NFT_TABLE}) || exitWithError "Failed to create table ${NFT_TABLE_BRIDGE} ${NFT_TABLE}"
+    ip netns exec "$CNI_CONTAINERID" $(create_table ${NFT_TABLE_NETDEV} ${NFT_TABLE}) || exitWithError "Failed to create table ${NFT_TABLE_BRIDGE} ${NFT_TABLE}"
+
+    # create base chains
+    ip netns exec "$CNI_CONTAINERID" $(create_netdev_base_chain ${NFT_TABLE_NETDEV} ${NFT_TABLE} ${NFT_INGRESS_CHAIN} ${NFT_INGRESS_HOOK} ${CNI_IFNAME}) || exitWithError "Failed to create netdev base chain ${NFT_INGRESS_CHAIN}"
```

Comment:
here we need to make sure to also write to the bridge table ( I'm aware there's the bug where only ARP frames are matched ..).